File: 2836-neither-minimum-nor-maximum/neither-minimum-nor-maximum.py
class Solution(object):
    def findNonMinOrMax(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 2:
            return -1
        
        # Taking just the first 3 elements guarantees finding a value 
        # that is neither the absolute minimum nor maximum of the entire array.
        sub = sorted(nums[:3])
        return sub[1]

File: 2836-neither-minimum-nor-maximum/neither-minimum-nor-maximum.java
import java.util.Arrays;

class Solution {
    public int findNonMinOrMax(int[] nums) {
        if (nums.length < 3) {
            return -1;
        }

        // Take only the first 3 distinct elements
        int[] sub = {nums[0], nums[1], nums[2]};
        Arrays.sort(sub);

        // The middle element is neither the minimum nor the maximum
        return sub[1];
    }
}